app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -35,56 +37,49 @@ class Api::V2::RepositoriesController < Api::V2::ApiController
     param :product_id, :number, :desc => N_("ID of a product to show repositories of")
     param :environment_id, :number, :desc => N_("ID of an environment to show repositories in")
     param :content_view_id, :number, :desc => N_("ID of a content view to show repositories in")
+    param :available_to_content_view_id, :number, :desc => N_("ID of content view to show repositories not in")
     param :content_view_version_id, :number, :desc => N_("ID of a content view version to show repositories in")
     param :erratum_id, String, :desc => N_("UUID of an erratum to find repositories that contain the erratum")
     param :library, :bool, :desc => N_("show repositories in Library and the default content view")
     param :content_type, String, :desc => N_("limit to only repositories of this time")
     param :name, String, :desc => N_("name of the repository"), :required => false
+    param :available_for, String, :desc => N_("interpret specified object to return only Repositories that can be associated with specified object.  Only 'content_view' is supported."),
+          :required => false
     param_group :search, Api::V2::ApiController
     # rubocop:disable Metrics/MethodLength
     def index
-      options = sort_params
-      options[:load_records?] = true
-
-      repositories = Repository.where(:product_id => Product.readable.where(:organization_id => @organization.id))
-      repositories = repositories.where(:product_id => @product.id) if @product
-
-      if params[:content_view_id] && params[:environment_id]
-        version = ContentViewVersion.in_environment(params[:environment_id]).where(:content_view_id => params[:content_view_id])
-        repositories = repositories.where(:content_view_version_id => version)
-      elsif params[:content_view_id]
-        repositories = repositories
-                         .joins(:content_view_repositories)
-                         .where("#{ContentViewRepository.table_name}.content_view_id" => params[:content_view_id])
-      end
+      options = {:includes => [:gpg_key, :product, :environment]}
+      respond(:collection => scoped_search(index_relation.uniq, :name, :desc, options))
+    end
 
-      repositories = repositories.where(:content_type => params[:content_type]) if params[:content_type]
-      repositories = repositories.where(:name => params[:name]) if params[:name]
-      repositories = repositories.joins(:errata).where("#{Erratum.table_name}.uuid" => params[:erratum_id]) if params[:erratum_id]
+    def index_relation
+      query = Repository.where(:product_id => Product.readable.where(:organization_id => @organization.id))
```

Comment:
Changed to use Repository.readable, and updated .readable here:  https://github.com/Katello/katello/pull/5317